README.md

```diff
@@ -1,2 +1,36 @@
-# libdependency
-Library for dependency-management related functionality
+# Libdependency
+
+Package libdependency provides a library for buildpack authors to construct code to retrieve new versions and their associated metadata.
+
+It also contains various helpers and actions to assist the overall dependency management workflow.
+
+## Integration
+
+### Retrieval
+
+The retrieve subpackage has an entrypoint func called `NewMetadata` that takes in various information and functions.
+Buildpack authors will need to pass implementations for `VersionFetcherFunc` and `GenerateMetadataFunc` into the
+`NewMetadataFunc`. The job of `NewMetadataFunc` is to read in the workflow inputs, run the provided functions, and provide
+the appropriate outputs to the workflow.
+
+```go
+type VersionFetcherFunc func() ([]versionology.VersionFetcher, error)
+type GenerateMetadataFunc func(version versionology.VersionFetcher) (cargo.ConfigMetadataDependency, error)
```

Comment:
```suggestion
type GenerateMetadataFunc func(version versionology.VersionFetcher) (versionology.Dependency, error)
```

Reply:
Good catch. The PR hasn't even been merged and already the docs are out of date. Yikes. Will fix in another push.